code/game/machinery/turntable.dm

```diff
@@ -1,280 +0,0 @@
-			S.channel = 10
-			S.wait = 1
-			for(var/mob/M in world)
-				M << S
-				M.music = 0
-			playing = 0
-			var/area/A = src.loc.loc
-			for(var/obj/machinery/party/lasermachine/L in A)
-				L.turnoff()
-
-
-
-/obj/machinery/party/lasermachine
-	name = "laser machine"
-	desc = "A laser machine that shoots lasers."
-	icon = 'icons/effects/lasers2.dmi'
-	icon_state = "lasermachine"
-	anchored = 1
-	var/mirrored = 0
-
-/obj/effect/turntable_laser
-	name = "laser"
-	desc = "A laser..."
-	icon = 'icons/effects/lasers2.dmi'
-	icon_state = "laserred1"
-	anchored = 1
-	layer = 4
-
-/obj/item/lasermachine/New()
-	..()
-
-/obj/machinery/party/lasermachine/proc/turnon()
-	var/wall = 0
-	var/cycle = 1
-	var/area/A = get_area(src)
-	var/X = 1
-	var/Y = 0
-	if(mirrored == 0)
-		while(wall == 0)
-			if(cycle == 1)
-				var/obj/effect/turntable_laser/F = new(src)
-				F.x = src.x+X
-				F.y = src.y+Y
-				F.z = src.z
-				F.icon_state = "laserred1"
-				var/area/AA = get_area(F)
-				var/turf/T = get_turf(F)
-				if(T.density == 1 || AA.name != A.name)
-					qdel(F)
-					return
-				cycle++
-				if(cycle > 3)
-					cycle = 1
-				X++
-			if(cycle == 2)
-				var/obj/effect/turntable_laser/F = new(src)
-				F.x = src.x+X
-				F.y = src.y+Y
-				F.z = src.z
-				F.icon_state = "laserred2"
-				var/area/AA = get_area(F)
-				var/turf/T = get_turf(F)
-				if(T.density == 1 || AA.name != A.name)
-					qdel(F)
-					return
-				cycle++
-				if(cycle > 3)
-					cycle = 1
-				Y++
-			if(cycle == 3)
-				var/obj/effect/turntable_laser/F = new(src)
-				F.x = src.x+X
-				F.y = src.y+Y
-				F.z = src.z
-				F.icon_state = "laserred3"
-				var/area/AA = get_area(F)
-				var/turf/T = get_turf(F)
-				if(T.density == 1 || AA.name != A.name)
-					qdel(F)
-					return
-				cycle++
-				if(cycle > 3)
-					cycle = 1
-				X++
-	if(mirrored == 1)
-		while(wall == 0)
-			if(cycle == 1)
-				var/obj/effect/turntable_laser/F = new(src)
-				F.x = src.x+X
-				F.y = src.y-Y
-				F.z = src.z
-				F.icon_state = "laserred1m"
-				var/area/AA = get_area(F)
-				var/turf/T = get_turf(F)
-				if(T.density == 1 || AA.name != A.name)
-					qdel(F)
-					return
-				cycle++
-				if(cycle > 3)
-					cycle = 1
-				Y++
-			if(cycle == 2)
-				var/obj/effect/turntable_laser/F = new(src)
-				F.x = src.x+X
-				F.y = src.y-Y
-				F.z = src.z
-				F.icon_state = "laserred2m"
-				var/area/AA = get_area(F)
-				var/turf/T = get_turf(F)
-				if(T.density == 1 || AA.name != A.name)
-					qdel(F)
-					return
-				cycle++
-				if(cycle > 3)
-					cycle = 1
-				X++
-			if(cycle == 3)
-				var/obj/effect/turntable_laser/F = new(src)
-				F.x = src.x+X
-				F.y = src.y-Y
-				F.z = src.z
-				F.icon_state = "laserred3m"
-				var/area/AA = get_area(F)
-				var/turf/T = get_turf(F)
-				if(T.density == 1 || AA.name != A.name)
-					qdel(F)
-					return
-				cycle++
-				if(cycle > 3)
-					cycle = 1
-				X++
-
-
-
-/obj/machinery/party/lasermachine/proc/turnoff()
-	var/area/A = src.loc.loc
```

Comment:
`src.loc.loc`

What??!?!

Reply:
yeah....`loc.loc` is usually the area.

loc of a mob is usually a turf, and the loc of the turf is an area.

I'm sure you can break this somehow, pretty easily though.